packages/jetbrains-plugin/src/test/kotlin/com/mongodb/jbplugin/codeActions/impl/MongoDbRunQueryActionUiTest.kt

```diff
@@ -20,12 +21,15 @@ class MongoDbRunQueryActionUiTest {
         remoteRobot: RemoteRobot,
         url: MongoDbServerUrl,
     ) {
+        // This is our safety net to ensure we start with a clean slate
+        remoteRobot.openDatabaseToolWindow().removeAllDataSources()
+        remoteRobot.closeRightToolWindow()
         remoteRobot.ideaFrame().addDataSourceWithUrl(javaClass.simpleName, url)
     }
 
     @AfterEach
     fun tearDown(remoteRobot: RemoteRobot) {
-        remoteRobot.ideaFrame().cleanDataSources()
+        remoteRobot.openDatabaseToolWindow().removeAllDataSources()
```

Comment:
So this is the change that makes sure that our tearDown don't fail. The cleanDataSources will fail from time to time, depending on the modality state of the application. Performing the same action via UI is working much more reliably.